internal/clients/ec2/ec2_client.go

```diff
@@ -122,6 +122,25 @@ func (c *Client) ListInstanceTypes() ([]types.InstanceTypeInfo, error) {
 	return resp.InstanceTypes, nil
 }
 
+func (c *Client) ListInstanceTypesWithPaginator() ([]types.InstanceTypeInfo, error) {
+	log.Trace().Msg("Listing AWS EC2 instance types Using Paginator")
+
+	input := &ec2.DescribeInstanceTypesInput{MaxResults: aws.Int32(100)}
+	pag := ec2.NewDescribeInstanceTypesPaginator(c.ec2, input)
+
+	res := make([]types.InstanceTypeInfo, 0)
+	for pag.HasMorePages() {
+		resp, err := pag.NextPage(c.context)
+		if err != nil {
+			return nil, fmt.Errorf("cannot list instance types: %w", err)
+		}
+		res = append(res, resp.InstanceTypes...)
+	}
+	log.Debug().Msgf("Number of AWS EC2 instance types: %d", len(res))
```

Comment:
Can you make this a trace call and delete the one above? I think it's okay for a small function to leave one trace message, but two is a bit too much.